index.js

```diff
@@ -26,19 +29,30 @@ module.exports = function (src, dest, opts) {
 
 	opts = objectAssign({overwrite: true}, opts);
 
-	return new Promise(function startRead(resolve, reject) {
-		var read = fs.createReadStream(src);
-
-		read.on('error', function (err) {
-			reject(new CpFileError('cannot read from `' + src + '`: ' + err.message, err));
-		});
+	var progress = {
+		stat: null,
+		emitter: new EventEmitter()
+	};
```

Comment:
I've thought about that, but we have:
```js
cpFile(…) // returns the promise
  .on(…)  // but the listeners are added afterwards
```

Therefore, a workaround to avoid the stat call would be rather awkward IMO, but if you have an idea…